lib/discourse_api/api/categories.rb

```diff
@@ -75,12 +75,17 @@ def category(id)
         response[:body]['category']
       end
 
+      # TODO: Deprecated. Remove after 20210727
       def category_set_user_notification(args = {})
         category_id = args[:id]
         args = API.params(args)
           .required(:notification_level)
         post("/category/#{category_id}/notifications", args)
       end
+
+      def category_set_user_notification_level(category_id, notification_level)
+        post("/category/#{category_id}/notifications?notification_level=#{notification_level}")
```

Comment:
We don't want to use query params here.

We can just pass the hash in the post body:

```
post("/category/#{category_id}/notifications", args)
```